File: app/src/main/java/com/epro/fastdevtem/mvp/BasePresenterImpl.java
package com.epro.fastdevtem.mvp;


import java.lang.ref.WeakReference;

public class BasePresenterImpl<V extends BaseView> implements BasePresenter<V>{
    private V mView;

    private WeakReference<V> reference;

    @Override
    public void attachView(V view) {

//        mView=view;
        reference = new WeakReference<V>(view);
    }

    @Override
    public void detachView() {

//        mView=null;
        if (reference != null) {
            reference = null;
        }
    }

    public V getView(){
        return reference.get();
    }
}

File: app/src/main/java/com/epro/fastdevtem/mvp/BasePresenterImpl.kt
package com.epro.fastdevtem.mvp


import java.lang.ref.WeakReference

open class BasePresenterImpl<V : BaseView> : BasePresenter<V> {
    private val mView: V? = null

    private var reference: WeakReference<V>? = null

    override fun attachView(view: V) {
        reference = WeakReference(view)
    }

    override fun detachView() {
        if (reference != null) {
            reference = null
        }
    }

    val view: V?
        get() = reference?.get()
}
